openapi-mock/server/mock_controller.js

```diff
@@ -67,32 +85,43 @@ function registerLogger(app) {
     });
 }
 
-function createEndpoints(openApi_doc, api) {
-    if (api.hasOwnProperty("added_endpoints")) {
-        api.added_endpoints.forEach(function (point) {
-            var endpoint = yaml.safeLoad(fs.readFileSync(point.filePath, 'utf8'));
-            if (!openApi_doc["paths"].hasOwnProperty(point.url)) {
-                LOGGER.debug("Adding custom endpoint %s to %s", point.url, api.name)
-                openApi_doc["paths"][point.url] = endpoint;
-                var yml_format = pretty_yaml.stringify(openApi_doc);
-                utility.writeToFile(api.specification_file, yml_format, true);
-            }
-        });
-    }
+function createEndpoints(openApi_doc, api, file_name) {
+
+    api.added_endpoints.forEach(function (point) {
+        var endpoint = yaml.safeLoad(fs.readFileSync(point.filePath, 'utf8'));
+        if (!openApi_doc["paths"].hasOwnProperty(point.url)) {
+            LOGGER.debug("Adding custom endpoint %s to %s", point.url, api.name)
+            openApi_doc["paths"][point.url] = endpoint;
+        }
+    });
+    var yml_format = pretty_yaml.stringify(openApi_doc);
+    utility.writeToFile(file_name, yml_format, true);

```

Comment:
Please restructure so that the file gets written once only. Read, add all endpoints, write. Everything else is confusing to me.